### PR TITLE
[tests] Improve performance of writing data for logs test

### PR DESCRIPTION
**Pull request by TurboTurtle**
The stagetwo logs test that verifies our journal size limiting is functional has long suffered from performance issues. This was due to an inefficient method of generating random data to write to the journal, in addition to compression algorithms generally having a difficult time with large amounts of random data.

Address this primarily by improving the method used to generate the random data in the first place. Rather than rebuilding a long string from via the `random.choice()` method against a list of alphanumeric characters, generate the random strings via the `random.getrandbits()` method for a fixed size, then convert that to a hex string.

Further, write the data in a more efficient manner by not exceeding the journal's default log threshold for limiting/throttling, instead regenerating and writing the random data in fixed-sized chunks. Additionally, only write enough data to surpass our size threshold.

As a side effect of this change, our standard compression approach for LZMA (.xz) was struggling due to the more random nature of the written data. Our preset override of 3, which typically would reduce system strain, was actually hindering the ability to compress the final archive in a reasonable amount of time. Increasing this preset to 6 alleviates this issue, while keeping the resource consumption of compressing the archive similar to that of an end user manually running the `xz` command.

Related: #2700

Signed-off-by: Jake Hunsaker <jhunsake@redhat.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

File: tests/report_tests/plugin_tests/logs.py [hunk 1]
```diff
@@ -48,20 +48,27 @@ class LogsSizeLimitTest(StageTwoReportTest):
...
-            sleep(10)
-            sosfd.write(rand[::-1] + '\n')
+        # journald default threshold size is 64k
+        rsize = 65535
+        to_write = ((101 * 1048576) - (j.get_usage()))
+        if to_write / 1048576 < 1:
+            return
+        try:
+            orig_limit = sys.get_int_max_str_digits()
+            sys.set_int_max_str_digits(rsize)
```
[the remark this answers is not in the document]
Comment (TurboTurtle) [reply tests/report_tests/plugin_tests/logs.py:63]: Yep - didn't realize it was a backport at first. Older pythons we should just be able to fail gracefully here with a nested try block.

**Comment by pmoravec** [comment]
I am all for improving performance, but either my tests or results are wrong (more probable), or I see no big benefit..?

1) `xz` using (default) `-6` compression - which should work well, indeed, but checking on a small sosreport tarball:

```
    # ll sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar
    -rw-r--r--. 1 root root 507658240 Dec 12 21:28 sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar
    # time xz -3 sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar
     
    real	0m27.358s
    user	0m25.954s
    sys	0m1.050s
    # ll sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar.xz 
    -rw-r--r--. 1 root root 23100924 Dec 12 21:28 sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar.xz
    # time xz -d sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar.xz 
    
    real	0m3.737s
    user	0m2.892s
    sys	0m0.776s
    # time xz -6 sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar
    
    real	1m44.462s
    user	1m42.109s
    sys	0m1.175s
    # ll sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar.xz 
    -rw-r--r--. 1 root root 20301064 Dec 12 21:28 sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar.xz
    # time xz -d sosreport-pmoravec-rhel8-2022-12-12-boecgus.tar.xz 
    
    real	0m3.739s
    user	0m2.754s
    sys	0m0.894s
    #
```

So we gain 15% disk space improvement for more than triple time..? (which is still low compared to `sos report` collection time)

2) Are the counts for journal logs generation correct? Since mimicking the code on stdout with "no journal so far":

```
    # cat new_rand.py
    import random
    from string import ascii_uppercase, digits
    from time import sleep
    import sys
     
    rsize = 65535
    to_write = 101 * 1048576
    try:
        orig_limit = sys.get_int_max_str_digits()
        sys.set_int_max_str_digits(rsize)
        for i in range(int(to_write / rsize) + 1):
            rand = f"{random.getrandbits(rsize * 8):x}"
            print(rand)
            sleep(0.1)
    except Exception:
        raise
    finally:
        sys.set_int_max_str_digits(orig_limit)
     
    # time python3 new_rand.py | wc
       1617    1617 211941697
     
    real	2m44.777s
    user	0m2.746s
    sys	0m0.211s
    #
```

I get 200MB data generated (current code generates 125829144b in 1:48 which is slower per byte - just the length is weird / doubled in the proposal..?)

**Comment by TurboTurtle** [comment]
That was me forgetting the halve the number of random bits generated when deciding to convert the bits to hex characters instead of just 0-9, to get a greater variety of available characters to write to the journal for this.

Fixed that, repushed.

**Comment by TurboTurtle** [comment]
So, this looks on the CI VMs to simply be transferring the runtime from the random data generation to the compression time within sos. On my local system, using `3` for lzma (xz) compression, the report takes a significantly longer time to compress at the end than with `6`, so I'm not sure if this is actually a meaningful change - we're just creating random data faster, but it happens to be harder to compress quickly.

**Comment by TurboTurtle** [comment]
Closing this for now. Will revisit later with another approach to get our compression time down.

**Pull request closed without merging**